cmd/examples/server.cpp

```diff
@@ -539,59 +541,61 @@ class MyServer : public quicr::Server
     }
+
     void OnFetchOk(quicr::ConnectionHandle connection_handle,
                    uint64_t subscribe_id,
                    const quicr::FullTrackName& track_full_name,
                    const quicr::FetchAttributes& attrs) override
     {
         auto pub_track_h =
-          std::make_shared<MyPublishTrackHandler>(track_full_name, quicr::TrackMode::kStream, attrs.priority, 5000);
+          std::make_shared<MyPublishTrackHandler>(track_full_name, quicr::TrackMode::kStream, attrs.priority, 50000);
         BindPublisherTrack(connection_handle, subscribe_id, pub_track_h);
 
         const auto th = quicr::TrackHash(track_full_name);
-        const auto& cache_entry = cache_.at(th.track_namespace_hash);
-
-        std::thread retrieve_cache_thread([=, &cache_entry] {
-            std::for_each(cache_entry.begin(), cache_entry.end(), [&](const auto& e) {
-                const auto& [group_id, object] = e;
-
-                if ((group_id < attrs.start_group || group_id >= attrs.end_group) ||
-                    (object.object_id < attrs.start_object || object.object_id >= attrs.end_object))
-                    return;
-
-                quicr::ObjectHeaders obj_headers{
-                    group_id,
-                    object.object_id,
-                    object.subgroup_id,
-                    object.data.size(),
-                    quicr::ObjectStatus::kAvailable,
-                    object.priority,
-                    object.ttl,
-                    std::nullopt,
-                    object.extensions,
-                };
-
-                pub_track_h->PublishObject(obj_headers, object.data);
-                std::this_thread::sleep_for(std::chrono::milliseconds(1));
-            });
-        });
+
+        std::thread retrieve_cache_thread(
+          [=, cache_entries = cache_.at(th.track_namespace_hash).Get(attrs.start_group, attrs.end_group)] {
+              for (const auto& cache_entry : cache_entries) {
+                  for (const auto& object : cache_entry) {
+                      if ((object.group_id < attrs.start_group || object.group_id >= attrs.end_group) ||
+                          (object.object_id < attrs.start_object || object.object_id >= attrs.end_object))
+                          continue;
+
+                      quicr::ObjectHeaders obj_headers{
+                          object.group_id,
+                          object.object_id,
+                          object.subgroup_id,
+                          object.data.size(),
+                          quicr::ObjectStatus::kAvailable,
+                          object.priority,
+                          object.ttl,
+                          std::nullopt,
+                          object.extensions,
+                      };
+
+                      pub_track_h->PublishObject(obj_headers, object.data);
+                      std::this_thread::sleep_for(std::chrono::milliseconds(1));
+                  }
+              }
+          });
```

Comment:
Not related to this change, but we need to clean up the bind publish track.  I have opened https://github.com/Quicr/libquicr/issues/332 to track that.

Reply:
Thanks for opening that!